include/audio/effects.h

```diff
@@ -0,0 +1,117 @@
+#ifndef AUDIO_EFFECTS_H
+#define AUDIO_EFFECTS_H
+
+#include "PR/ultratypes.h"
+#include "unk.h"
+
+struct Note;
+struct SequencePlayer;
+
+/* Multi-Point ADSR */
+
+typedef enum AdsrStatus {
```

Comment:
What's an `ADSR`?

Reply:
It stands for **Attack, Decay, Sustain, Release**, it describes how you play a note's volume over the entire duration of a note:
https://en.wikipedia.org/wiki/Envelope_(music)
https://www.masterclass.com/articles/adsr-envelope-explained

Reply:
I'd also like to add for any audio enthusiasts who may stumble on this PR, a todo outside the scope of this PR is to address that ADSR might not be the best name for this process. SM64/OoT/MM/AF all treat this volume envelope as more of a script with commands that can make a custom shape (to what extent I haven't explored fully) but it's not really limited to the ADSR shape you commonly see:

![image](https://github.com/zeldaret/mm/assets/47598039/ed31d232-ad99-459a-8f40-aeba0d820134)

Probably the best thing is to just stick with `ADSR` -> `Envelope` in all cases, and I think this was starting to be address in the audio extraction PR. But I'd like to explore Envelopes more before making that change. Some of the names like `delay` and `release` in master for adsr may need tweaking too. Soon :tm: